File: frontend/src/api/articleController.ts
// @ts-ignore
/* eslint-disable */
import request from "@/ui-backend/utils";

/** 此处后端没有提供注释 PUT /backend/article/ */
export async function updateArticle(
  body: API.ArticleVO,
  options?: { [key: string]: any }
) {
  return request<API.BaseResponseBoolean>("/backend/article/", {
    method: "PUT",
    headers: {
      "Content-Type": "application/json",
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /backend/article/ */
export async function addArticle(
  body: API.ArticleVO,
  options?: { [key: string]: any }
) {
  return request<API.BaseResponseBoolean>("/backend/article/", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 DELETE /backend/article/ */
export async function deleteArticle(
  body: API.DeleteRequest,
  options?: { [key: string]: any }
) {
  return request<API.BaseResponseBoolean>("/backend/article/", {
    method: "DELETE",
    headers: {
      "Content-Type": "application/json",
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /backend/article/${param0} */
export async function getArticleVoById(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.getArticleVOByIdParams,
  options?: { [key: string]: any }
) {
  const { id: param0, ...queryParams } = params;
  return request<API.BaseResponseArticleVO>(`/backend/article/${param0}`, {
    method: "GET",
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /backend/article/list */
export async function getArticleVoList(options?: { [key: string]: any }) {
  return request<API.BaseResponseListArticleVO>("/backend/article/list", {
    method: "GET",
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /backend/article/list/page/vo */
export async function getArticleVoListByPage(
  body: API.ArticleQueryRequest,
  options?: { [key: string]: any }
) {
  return request<API.BaseResponsePageArticleVO>(
    "/backend/article/list/page/vo",
    {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      data: body,
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 POST /backend/article/upload/image */
export async function getUploadFile(
  body: {},
  options?: { [key: string]: any }
) {
  return request<API.BaseResponseString>("/backend/article/upload/image", {
    method: "POST",
    // headers: {
    //   "Content-Type": "application/json",
    // },
    data: body,
    ...(options || {}),
  });
}

File: src/main/java/org/example/blogsakuraDDD/interfaces/controller/ArticleController.java
package org.example.blogsakuraDDD.interfaces.controller;

import com.mybatisflex.core.paginate.Page;
import jakarta.annotation.Resource;
import jakarta.servlet.http.HttpServletRequest;
import lombok.extern.slf4j.Slf4j;
import org.example.blogsakuraDDD.infrastruct.aop.Log;
import org.example.blogsakuraDDD.infrastruct.common.BaseResponse;
import org.example.blogsakuraDDD.infrastruct.common.DeleteRequest;
import org.example.blogsakuraDDD.infrastruct.common.ResultUtils;
import org.example.blogsakuraDDD.interfaces.dto.blog.article.ArticleQueryRequest;
import org.example.blogsakuraDDD.interfaces.vo.blog.article.ArticleVO;
import org.springframework.web.bind.annotation.*;
import org.example.blogsakuraDDD.application.service.ArticleApplicationService;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

/**
 * 文章表 控制层。
 *
 * @author <a href="https://github.com/liuweixu">liuweixu</a>
 */
@RestController
@RequestMapping("/backend/article")
@Slf4j
public class ArticleController {

    @Resource
    private ArticleApplicationService articleApplicationService;

    /**
     * 新增文章表。
     *
     * @param articleVO 前端返回的文章表
     * @return {@code true} 保存成功，{@code false} 保存失败
     */
    @PostMapping("/")
    @Log("新增文章")
    public BaseResponse<Boolean> addArticle(@RequestBody ArticleVO articleVO) {
        return ResultUtils.success(articleApplicationService.addArticle(articleVO));
    }

    /**
     * 根据主键删除文章表。
     *
     * @param deleteRequest 删除请求
     * @return {@code true} 删除成功，{@code false} 删除失败
     */
    @DeleteMapping("/")
    @Log("删除文章")
    public BaseResponse<Boolean> deleteArticle(@RequestBody DeleteRequest deleteRequest, HttpServletRequest request) {
        return ResultUtils.success(articleApplicationService.deleteArticle(deleteRequest, request));
    }

    /**
     * 根据主键更新文章表。
     *
     * @param articleVO 前端返回的文章表
     * @return {@code true} 更新成功，{@code false} 更新失败
     */
    @PutMapping("/")
    @Log("更新文章")
    public BaseResponse<Boolean> updateArticle(@RequestBody ArticleVO articleVO) {
        return ResultUtils.success(articleApplicationService.updateArticle(articleVO));
    }

    /**
     * 查询所有文章表。
     *
     * @return 所有数据
     */
    @GetMapping("list")
    public BaseResponse<List<ArticleVO>> getArticleVOList() {
        return ResultUtils.success(articleApplicationService.getArticleVOList());
    }

    /**
     * 根据主键获取文章。
     *
     * @param id 文章表主键
     * @return 文章表详情
     */
    @GetMapping("/{id}")
    public BaseResponse<ArticleVO> getArticleVOById(@PathVariable Long id) {
        return ResultUtils.success(articleApplicationService.getArticleVOById(id));
    }

    /**
     * 分页查询文章表。
     *
     * @param articleQueryRequest 分页查询请求
     * @return 分页对象
     */
    @PostMapping("list/page/vo")
    public BaseResponse<Page<ArticleVO>> getArticleVOListByPage(@RequestBody ArticleQueryRequest articleQueryRequest) {
        return ResultUtils.success(articleApplicationService.getArticleVOListByPage(articleQueryRequest));
    }

    @PostMapping("/upload/image")
    public BaseResponse<String> getUploadFile(@RequestPart("image") MultipartFile file) {
        // 获取文件名
        return ResultUtils.success(articleApplicationService.getUploadFile(file));
    }

}
